src/testRunner/unittests/sys/symlinkWatching.ts

```diff
@@ -45,10 +45,15 @@ describe("unittests:: sys:: symlinkWatching::", () => {
                     watcher: sys.watchFile!(
                         toWatch,
                         (fileName, eventKind, modifiedTime) => {
-                            assert.equal(fileName, toWatch);
-                            assert.equal(eventKind, ts.FileWatcherEventKind.Changed);
-                            const actual = modifiedTimeToString(modifiedTime);
-                            assert(actual === undefined || actual === modifiedTimeToString(sys.getModifiedTime!(file)));
+                            try {
+                                assert.equal(fileName, toWatch);
+                                assert.equal(eventKind, ts.FileWatcherEventKind.Changed);
+                                const actual = modifiedTimeToString(modifiedTime);
+                                assert(actual === undefined || actual === modifiedTimeToString(sys.getModifiedTime!(file)));
+                            }
+                            catch (e) {
+                                result.deferred.reject(e);
+                            }

```

Comment:
I think this may need a return, otherwise it will fall through to the resolve, right?

Reply:
(note: resolving a rejected promise is a noop)